scripts/compatibility_scanner/compatibility_scanner.q

```diff
@@ -0,0 +1,239 @@
+  checksDict,:enlist[`system_command]!enlist readCommandsChecks COMMANDS_CHECKS_TSV;
+
+  if[DEBUG_SHOW_REGEX_PASSED;
+    -1"DEBUG: Regex pattern being passed:";
+    -1 combineRegexList value checksDict;
+  ];
+  
+  :checksDict;
+ };
+
+readCommandsChecks:{[file]
+  tbl:("**";enlist"\t") 0: hsym`$file;
+
+  :combineRegexList tbl`regex;
+ };
+
+readAssignmentChecks:{[file]
+  tbl:("**";enlist"\t") 0: hsym`$file;
+
+  :combineRegexList raze{
+    :enlist[y] cross x cross enlist z;
+  }[ZS_TO_CHECK]'[tbl`prefix;tbl`suffix];
+ };
+
+combineRegexList:{[checksList]
+  :-1 _ raze{"(",x,")|"}each checksList;
+ };
+
+filterExcluded:{[files;excludedFiles]
+  :files where not files in excludedFiles;
+ };
+
+writeToCsv:{[file;checksDict;lines]
+  file:getFullPath file;
+  if[not file like "*[.]csv";
+    -1"WARN: Adding '.csv' suffix to file '",file,"'";
+    file,:".csv";
+  ];
+
+  tbl:categoriseLines[checksDict;lines];
+
+  -1"Saving to file '",file,"' . . .";
+  @[{(hsym`$x) 0: csv 0: y;-1"Saved"}[file];tbl;
+    {[x;y] -2"ERROR: Could not save to '",x,"' due to:\n",y}[file]];
+ };
+
+categoriseLines:{[checksDict;lines]
+  -1"Categorising lines . . .";
+
+  tbl:raze categoriseLine[checksDict]each lines;
+
+  -1"Categorised lines";
+
+  :tbl;
+ };
+
+categoriseLine:{[checkDict;line]
+  .progressTracker.dotNum:@[{mod[1+value x;4]};`.progressTracker.dotNum;0];
+  1"Categorising ",.progressTracker.dotNum#".";  // Used to give the user a visual indication that the program is still progressing without clogging up the logs
+
+  res:raze{[check;checkRegex;line]
+    lineDict:{`file`lineNum`code!(x 0;"J"$x 1;":" sv 2 _ x)}":" vs line;
+
+    found:0<"J"$first system"printf '%s' $(echo '",ssr[lineDict[`code];"'";"'\\''"],"' | grep -cE '",checkRegex,"' )";
+    if[not found;:()];
+
+    :([]
+      category:enlist check;
+      file:enlist lineDict`file;
+      line_num:enlist lineDict`lineNum;
+      code:enlist lineDict`code
+    );
+  }[;;line]'[key checkDict;value checkDict];
+
+  1"\033[2K\r";  // Clearing the line for the next step to print over it (ANSI escape code to erase current line + carriage return)
+
+  :res;
+ };
+
+showHelp:{[]
+  -1"+--------------------------------+";
+  -1"| FinSpace Compatibility Scanner |";
+  -1"+--------------------------------+";
+  -1"[Arguments]";
+  -1"";
+  -1"-dir:      Directory to scan";
+  -1"-file:     In normal mode: Scans file(s);";
+  -1"           In regex mode:  Scans file(s) matching regex in dir";
+  -1"                           (or from `pwd` if no dir is passed)";
+  -1"-exclude:  In normal mode: Excludes file(s) from scan;";
+  -1"           In regex mode:  Excludes file(s) from scan matching regex in dir";
+  -1"                           (or from `pwd` if no dir is passed)";
+  -1"-csv:      CSV file to output the results to";
+  -1"           (Adds '.csv' to the end of the filename if not provided)";
+  -1"--regex:   Switch file and exclude arguments to regex mode (posix-extended)";
```

Comment:
Assuming without this flag the script is considered in "normal mode"?

Reply:
Yea, I'll add an extra bit mentioning that in the help menu